config/src/keys.rs

```diff
@@ -2,19 +2,14 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::trusted_peers::{deserialize_key, serialize_key};
-use libra_crypto::{
-    ed25519::*,
-    test_utils::TEST_SEED,
-    x25519::{self, X25519StaticPrivateKey, X25519StaticPublicKey},
-    PrivateKey, ValidKeyStringExt,
-};
+use libra_crypto::{test_utils::TEST_SEED, PrivateKey, Uniform, ValidKeyStringExt};
 use mirai_annotations::verify_unreachable;
 use rand::{rngs::StdRng, SeedableRng};
 use serde::{de::DeserializeOwned, Deserialize, Deserializer, Serialize, Serializer};
 
 #[derive(Debug, PartialEq)]
 #[cfg_attr(any(test, feature = "fuzzing"), derive(Clone))]
-enum PrivateKeyContainer<T> {
+pub enum PrivateKeyContainer<T> {
```

Comment:
Does this need to be public?

Reply:
Compiler yelled at me, apparently due to KeyPair

warning: private type `keys::PrivateKeyContainer<T>` in public interface (error E0446)
  --> config/src/keys.rs:68:17
   |
68 | #[derive(Debug, Deserialize, PartialEq, Serialize)]
   |                 ^^^^^^^^^^^
   |
   = note: `#[warn(private_in_public)]` on by default
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>